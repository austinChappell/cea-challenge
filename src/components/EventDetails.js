import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router';

import Api from '../api/api';

const api = new Api();

const { getMeetupData } = api;

class EventDetails extends Component {
  state = {
    authorized: false,
  }

  componentWillMount() {
    this.loadPage();
  }

  loadData = (data) => {
    console.log('DATA', data)
  }

  loadPage = () => {
    const { accessToken } = this.props;
    console.log('ACCESS TOKEN', accessToken)
    const authorized = accessToken ? true : false;
    if (authorized) {
      getMeetupData('reactjs-dallas', accessToken, this.loadData)
    }
    this.setState({ authorized })
  }

  render() {
    const authCheck = this.state.authorized ? null : <Redirect to="/" />

    return (
      <div className="EventDetails">
        {authCheck}
        Event Details Component
      </div>
    )
  }
} 

const mapStateToProps = state => ({
  accessToken: state.generalReducer.accessToken,
});

export default connect(mapStateToProps)(EventDetails);
