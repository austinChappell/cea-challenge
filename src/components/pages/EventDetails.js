import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment-timezone';

import actions from '../../store/actions';
import Api from '../../api/api';
import mockEventData from '../../data/event_data';

import Banner from '../shared/Banner';
import Card from '../shared/Card';
import RouteRestrictor from '../shared/RouteRestrictor';
import SliderCarousel from '../shared/SliderCarousel';

const api = new Api();

const { getMeetupEvent } = api;

const propTypes = {
  accessToken: PropTypes.string,
  match: PropTypes.objectOf(PropTypes.any).isRequired,
  selectedEvent: PropTypes.objectOf(PropTypes.any),
  setAccessToken: PropTypes.func.isRequired,
};

const defaultProps = {
  accessToken: null,
  selectedEvent: null,
};

class EventDetails extends Component {
  state = {
    rsvps: [],
  }

  componentWillMount() {
    const storageToken = sessionStorage.getItem('accessToken');
    if (!this.props.accessToken && storageToken) {
      this.props.setAccessToken(storageToken);
    }
  }

  componentDidMount() {
    const { accessToken, match } = this.props;
    const { eventId, groupName } = match.params;
    if (process.env.REACT_APP_ENV === 'development') {
      this.setEventData(mockEventData);
    } else {
      getMeetupEvent(groupName, eventId, accessToken, this.setEventData);
    }
    window.scroll(0, 0);
  }

  setEventData = (results) => {
    const rsvps = results.map((r => r.member));
    this.setState({ rsvps });
  }

  render() {
    const { selectedEvent } = this.props;
    const { name } = selectedEvent;
    const countDownMessage = `Until the ${name} Meetup`;

    return (
      <div className="EventDetails">
        <RouteRestrictor />
        <Banner
          countDownMessage={countDownMessage}
          nextEventTime={selectedEvent.time}
          subTitle={selectedEvent.name}
        />
        <div className="details">
          <div>
            <h4>Where:</h4>
            <div>
              <p>
                {selectedEvent.venue ? selectedEvent.venue.name : 'TBD'}
              </p>
            </div>
          </div>
          <div>
            <h4>When:</h4>
            <div>
              <p>
                {moment.tz(selectedEvent.time, 'America/Chicago').format('MMMM Do, YYYY, h:mm a')}
              </p>
            </div>
          </div>
          <div>
            <h4>Details:</h4>
            <div dangerouslySetInnerHTML={{ __html: selectedEvent.description }} />
          </div>
          <SliderCarousel title="RSVP">
            {this.state.rsvps.map(member => (
              <Card key={member.id}>
                <div>
                  <div className="item">
                    <div
                      className="avatar"
                      style={{ backgroundImage: `url(${member.photo.thumb_link})` }}
                    />
                  </div>
                  <h6>{member.name}</h6>
                </div>
              </Card>
            ))}
          </SliderCarousel>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  accessToken: state.generalReducer.accessToken,
  selectedEvent: state.eventReducer.selectedEvent,
});

const mapDispatchToProps = dispatch => ({
  setAccessToken: (accessToken) => {
    dispatch(actions.setAccessToken(accessToken));
  },
});

EventDetails.propTypes = propTypes;
EventDetails.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(EventDetails);
