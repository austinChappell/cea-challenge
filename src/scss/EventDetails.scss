.EventDetails {
  .details {
    padding: 30px;
    & > div {
      margin: 10px 0;
      h4 {
        margin-bottom: 5px;
      }
    }
  }
}