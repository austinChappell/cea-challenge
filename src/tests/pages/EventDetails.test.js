import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import EventDetails from '../../components/pages/EventDetails';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  generalReducer: {
    accessToken: 'fakeToken',
  },
  eventReducer: {
    selectedEvent: {},
  },
};

const props = {
  match: {
    params: {},
  },
};

describe('The Panel Component', () => {
  it('should render correctly', () => {
    const options = {
      context: {
        store: mockStore(initialState),
      },
    };
    const wrapper = shallow(<EventDetails {...props} />, options);
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
