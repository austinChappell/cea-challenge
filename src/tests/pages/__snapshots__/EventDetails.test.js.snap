// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The EventDetails Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventDetails
    accessToken="fakeToken"
    match={
      Object {
        "params": Object {},
      }
    }
    selectedEvent={
      Object {
        "time": 1522030848733,
      }
    }
    setAccessToken={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(RouteRestrictor) />,
        <Banner
          countDownMessage="Until the undefined Meetup"
          goBackButton={false}
          goBackRoute={null}
          goBackText={null}
          nextEventTime={1522030848733}
          subTitle={null}
          title={null}
        />,
        <div
          className="details"
        >
          <div>
            <h4>
              Where:
            </h4>
            <div>
              <p>
                TBD
              </p>
            </div>
          </div>
          <div>
            <h4>
              When:
            </h4>
            <div>
              <p>
                March 25th, 2018, 9:20 pm
              </p>
            </div>
          </div>
          <div>
            <h4>
              Details:
            </h4>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
            />
          </div>
          <SliderCarousel
            title="RSVP"
          />
        </div>,
      ],
      "className": "EventDetails",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "countDownMessage": "Until the undefined Meetup",
          "goBackButton": false,
          "goBackRoute": null,
          "goBackText": null,
          "nextEventTime": 1522030848733,
          "subTitle": null,
          "title": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h4>
                Where:
              </h4>
              <div>
                <p>
                  TBD
                </p>
              </div>
            </div>,
            <div>
              <h4>
                When:
              </h4>
              <div>
                <p>
                  March 25th, 2018, 9:20 pm
                </p>
              </div>
            </div>,
            <div>
              <h4>
                Details:
              </h4>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
            </div>,
            <SliderCarousel
              title="RSVP"
            />,
          ],
          "className": "details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Where:
                </h4>,
                <div>
                  <p>
                    TBD
                  </p>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Where:",
                },
                "ref": null,
                "rendered": "Where:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    TBD
                  </p>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TBD",
                  },
                  "ref": null,
                  "rendered": "TBD",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  When:
                </h4>,
                <div>
                  <p>
                    March 25th, 2018, 9:20 pm
                  </p>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "When:",
                },
                "ref": null,
                "rendered": "When:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    March 25th, 2018, 9:20 pm
                  </p>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "March 25th, 2018, 9:20 pm",
                  },
                  "ref": null,
                  "rendered": "March 25th, 2018, 9:20 pm",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Details:
                </h4>,
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": undefined,
                    }
                  }
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Details:",
                },
                "ref": null,
                "rendered": "Details:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "dangerouslySetInnerHTML": Object {
                    "__html": undefined,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [],
              "title": "RSVP",
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(RouteRestrictor) />,
          <Banner
            countDownMessage="Until the undefined Meetup"
            goBackButton={false}
            goBackRoute={null}
            goBackText={null}
            nextEventTime={1522030848733}
            subTitle={null}
            title={null}
          />,
          <div
            className="details"
          >
            <div>
              <h4>
                Where:
              </h4>
              <div>
                <p>
                  TBD
                </p>
              </div>
            </div>
            <div>
              <h4>
                When:
              </h4>
              <div>
                <p>
                  March 25th, 2018, 9:20 pm
                </p>
              </div>
            </div>
            <div>
              <h4>
                Details:
              </h4>
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
            </div>
            <SliderCarousel
              title="RSVP"
            />
          </div>,
        ],
        "className": "EventDetails",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "countDownMessage": "Until the undefined Meetup",
            "goBackButton": false,
            "goBackRoute": null,
            "goBackText": null,
            "nextEventTime": 1522030848733,
            "subTitle": null,
            "title": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h4>
                  Where:
                </h4>
                <div>
                  <p>
                    TBD
                  </p>
                </div>
              </div>,
              <div>
                <h4>
                  When:
                </h4>
                <div>
                  <p>
                    March 25th, 2018, 9:20 pm
                  </p>
                </div>
              </div>,
              <div>
                <h4>
                  Details:
                </h4>
                <div
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": undefined,
                    }
                  }
                />
              </div>,
              <SliderCarousel
                title="RSVP"
              />,
            ],
            "className": "details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Where:
                  </h4>,
                  <div>
                    <p>
                      TBD
                    </p>
                  </div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Where:",
                  },
                  "ref": null,
                  "rendered": "Where:",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      TBD
                    </p>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TBD",
                    },
                    "ref": null,
                    "rendered": "TBD",
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    When:
                  </h4>,
                  <div>
                    <p>
                      March 25th, 2018, 9:20 pm
                    </p>
                  </div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "When:",
                  },
                  "ref": null,
                  "rendered": "When:",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p>
                      March 25th, 2018, 9:20 pm
                    </p>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "March 25th, 2018, 9:20 pm",
                    },
                    "ref": null,
                    "rendered": "March 25th, 2018, 9:20 pm",
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Details:
                  </h4>,
                  <div
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": undefined,
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Details:",
                  },
                  "ref": null,
                  "rendered": "Details:",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "dangerouslySetInnerHTML": Object {
                      "__html": undefined,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "title": "RSVP",
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
